metrics/metrics.go

```diff
@@ -92,28 +123,106 @@ func watchMetrics(
 	ticker.Start(false, true, true)
 }
 
-func watchRawMetrics(
-	client *client.Client,
-	source RawSource,
+func watchMetricsProm(
+	c *client.Client,
+	sources map[string]Source,
 	interval time.Duration,
 ) {
-	metricsPipe := make(chan RawMetrics)
-	go sendRawMetrics(client, metricsPipe)
-	defer close(metricsPipe)
 
-	ticker := utils.NewTicker("raw-metrics", interval, func() {
-		metrics, err := source.GetRawMetrics()
-		if err != nil {
-			client.Errorf(err, "unable to retrieve metrics from sink")
-		}
+	ticker := utils.NewTicker(
+		"prom-metrics",
+		interval,
+		func(tickTime time.Time) {
+			m := sync.Mutex{}
+			metricsBatch := &MetricsBatch{
+				Timestamp: tickTime,
```

Comment:
Separation of timestamps across multiple sources can avoid rate error due to the lag of metrics retrieval for different sources.